src/IB/IBFEMethod.cpp

```diff
@@ -402,163 +417,71 @@ IBFEMethod::IBFEMethod(const std::string& object_name,
-        d_registered_for_restart = false;
-    }
-    return;
-} // ~IBFEMethod
-
 FEDataManager*
 IBFEMethod::getFEDataManager(const unsigned int part) const
 {
     TBOX_ASSERT(d_fe_equation_systems_initialized);
-    TBOX_ASSERT(part < d_num_parts);
+    TBOX_ASSERT(part < d_meshes.size());
     return d_primary_fe_data_managers[part];
 } // getFEDataManager
 
 void
 IBFEMethod::registerStressNormalizationPart(unsigned int part)
 {
     TBOX_ASSERT(d_fe_equation_systems_initialized);
-    TBOX_ASSERT(part < d_num_parts);
-    if (d_is_stress_normalization_part[part]) return;
+    TBOX_ASSERT(part < d_meshes.size());
+    if (d_stress_normalization_part[part]) return;
     d_has_stress_normalization_parts = true;
-    d_is_stress_normalization_part[part] = true;
-    auto& Phi_system = d_equation_systems[part]->add_system<LinearImplicitSystem>(PHI_SYSTEM_NAME);
+    d_stress_normalization_part[part] = true;
+    auto& Phi_system = d_equation_systems[part]->add_system<LinearImplicitSystem>(PRESSURE_SYSTEM_NAME);
     d_equation_systems[part]->parameters.set<Real>("Phi_epsilon") = d_epsilon;
     Phi_system.attach_assemble_function(assemble_poisson);
     // This system has a single variable so we don't need to also specify diagonal coupling
     Phi_system.add_variable("Phi", d_fe_order[part], d_fe_family[part]);
-    return;
-} // registerStressNormalizationPart
-
-void
-IBFEMethod::registerInitialCoordinateMappingFunction(const CoordinateMappingFcnData& data, const unsigned int part)
-{
-    TBOX_ASSERT(part < d_num_parts);
-    d_coordinate_mapping_fcn_data[part] = data;
-    return;
-} // registerInitialCoordinateMappingFunction
-
-IBFEMethod::CoordinateMappingFcnData
-IBFEMethod::getInitialCoordinateMappingFunction(unsigned int part) const
-{
-    TBOX_ASSERT(part < d_num_parts);
-    return d_coordinate_mapping_fcn_data[part];
-} // getInitialCoordinateMappingFunction
-
-void
-IBFEMethod::registerInitialVelocityFunction(const InitialVelocityFcnData& data, const unsigned int part)
-{
-    TBOX_ASSERT(part < d_num_parts);
-    d_initial_velocity_fcn_data[part] = data;
-    return;
-} // registerInitialVelocityFunction
-
-IBFEMethod::InitialVelocityFcnData
-IBFEMethod::getInitialVelocityFunction(unsigned int part) const
-{
-    TBOX_ASSERT(part < d_num_parts);
-    return d_initial_velocity_fcn_data[part];
-} // getInitialVelocityFunction
-
-void
-IBFEMethod::registerPK1StressFunction(const PK1StressFcnData& data, const unsigned int part)
-{
-    TBOX_ASSERT(part < d_num_parts);
-    d_PK1_stress_fcn_data[part].push_back(data);
-    if (data.quad_type == INVALID_Q_RULE)
+    // Setup cached system vectors.
+    const std::vector<std::string> vector_names{ "half", "tmp", "RHS Vector" };
+    for (const std::string& vector_name : vector_names)
     {
-        d_PK1_stress_fcn_data[part].back().quad_type = d_default_quad_type[part];
+        convert_parallel_to_ghosted(vector_name, Phi_system);
```

Comment:
Could we move this into the restart data load section somehow instead? IIRC this problem only occurs when we load vectors, not when we create them.

Reply:
See above.